src/core/frame.c

```diff
@@ -264,6 +265,10 @@ static MVMFrame * allocate_frame(MVMThreadContext *tc, MVMStaticFrame *static_fr
     /* Allocate space for lexicals and work area. */
     static_frame_body = &(static_frame->body);
     env_size = spesh_cand ? spesh_cand->env_size : static_frame_body->env_size;
+
+    jitcode = spesh_cand ? spesh_cand->jitcode : NULL;
+    num_locals = jitcode && jitcode->local_types ? jitcode->num_locals :
+        (spesh_cand ? spesh_cand->num_locals : static_frame_body->num_locals);
```

Comment:
Any reason the `num_locals` in `spesh->num_locals` can't be updated, to save the conditional and extra deref on this hot path?